query/src/sessions/session.rs

```diff
@@ -168,6 +170,28 @@ impl Session {
         self.mutable_state.set_current_user(user)
     }
 
+    pub fn validate_privilege(
```

Comment:
check_privilege is better?

Reply:
there're some `validate_` and `check_` functions in our codebase, as the existing convention, the `check_` functions often return a bool and used in an if condition, and the `validate_` functions often returns an Result<()>, and mostly used as a short-circuit call with a `?`, which breaks the control flow if any error occurred, like:

```
validate_expression(&expr)?;
```

```
   57 fn validate_function_arg(
   58     name: &str,
   59     args_len: usize,
   60     variadic_arguments: Option<(usize, usize)>,
   61     num_arguments: usize,
   62 ) -> Result<()> {```

in this case, the `validate_privilege` function would return an Result<()> and used as a short-circuit call with a `?`, i guess naming it as `validate_` follows the convension in the current codebase? 🤔

Reply:
Good 👍